src/include/msgr.h

```diff
@@ -188,6 +190,7 @@ struct ceph_msg_footer {
 #define CEPH_MSG_FOOTER_COMPLETE  (1<<0)   /* msg wasn't aborted */
 #define CEPH_MSG_FOOTER_NOCRC     (1<<1)   /* no data crc */
 #define CEPH_MSG_FOOTER_SIGNED	  (1<<2)   /* msg was signed */
+#define CEPH_MSG_FOOTER_COMPRESS  (1<<3)   /* msg was compressed */
```

Comment:
I think a better way to approach this is a flag in the header, CEPH_MSG_HEADER_COMPRESSDATA and COMPRESSFRONT and COMPRESSMIDDLE (or whatever the naming scheme is).  And make the first 4 bytes of each of those payloads be the decompressed length.  This way we don't change any of the fundamental msg structures.